### PR TITLE
GEN-1216 | Display attention card after removing extension offer

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Display attention card with message after removing car trial offer extension

- Implement component to replace some placeholder inside a text string

## Justify why they are needed

- Updates to offer page design

- The new design actually uses a "ToastBar", however we haven't implemented that yet, so I've just used an attention card for now

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3282** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3282" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3282?utm_source=stack-comment).

File: apps/store/src/features/carDealership/TrialExtensionForm.tsx [hunk 1]
```diff
@@ -111,6 +122,23 @@ export const TrialExtensionForm = (props: Props) => {
   )
 }
 
+type ReplaceTextProps = TextProps & {
+  text: string
+  children: string
```
Comment (alebedev) [inline apps/store/src/features/carDealership/TrialExtensionForm.tsx:127]: Nit: Only accepting string children vs another string prop - what is more obvious for the caller?  No strong opinion from me

File: apps/store/src/features/carDealership/TrialExtensionForm.tsx [hunk 2]
```diff
@@ -111,6 +122,23 @@ export const TrialExtensionForm = (props: Props) => {
...
+}
+
+const ReplaceText = ({ text, children, ...props }: ReplaceTextProps) => {
+  const placeholder = '{}'
+  const [before, after] = children.split(placeholder, 2)
+  return (
+    <>
+      {before}
+      <Text {...props}>{text}</Text>
```
Comment (alebedev) [inline apps/store/src/features/carDealership/TrialExtensionForm.tsx:136]: Do we want replacement text to have different style?  Not clear why we wrap it separately

**Review by alebedev — APPROVED** [review]
Looks good, few questions

File: apps/store/src/features/carDealership/TrialExtensionForm.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/features/carDealership/TrialExtensionForm.tsx:127]: ahaaa, right yeah that is a good point -- might be overloading children-prop a little too much

This is kind of a hack -- let's see if it survives when we refactor the code

File: apps/store/src/features/carDealership/TrialExtensionForm.tsx (hunk 2 above)
Comment (robinandeer) [reply apps/store/src/features/carDealership/TrialExtensionForm.tsx:136]: So what I want to achieve is some substring to be styled differently from the rest of the text.

The rest of the text should pick up styles from a different parent component.

**Comment by robinandeer** [comment]
### Merge Activity

* **Sep 28, 2:31 PM**: @robinandeer started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3282).

**Pull request merged**